server/api/crud/runs.py

```diff
@@ -468,3 +455,59 @@ async def _post_delete_run(project, uid):
                 project,
                 uid,
             )
+
+    @staticmethod
+    def _get_artifacts_from_uris(
+        db_session: sqlalchemy.orm.Session, project: str, run: dict
+    ):
+        """Fetch run artifacts by their artifact URIs in the run status"""
+        artifacts = []
+        artifact_uris = run.get("status", {}).get("artifact_uris", {})
+        for _, uri in artifact_uris.items():
+            _, uri = mlrun.datastore.parse_store_uri(uri)
+            project, key, iteration, tag, producer_id = mlrun.utils.parse_artifact_uri(
+                uri, project
+            )
+            artifact = server.api.crud.Artifacts().get_artifact(
+                db_session,
+                key=key,
+                tag=tag,
+                iter=iteration,
+                project=project,
+                producer_id=producer_id,
+                raise_on_not_found=False,
+            )
+            if artifact:
+                artifacts.append(artifact)
+            else:
+                logger.warning(
+                    "Failed to fetch run artifact, it may have been deleted",
+                    project=project,
+                    key=key,
+                    tag=tag,
+                    iteration=iteration,
+                )
+        return artifacts
+
+    @staticmethod
+    def _list_run_artifacts(
+        db_session: sqlalchemy.orm.Session,
+        iteration: int,
+        producer_id: str,
+        project: str,
+    ):
+        best_iteration = False
+        # If the iteration is 0, we mark the artifacts as best iteration.
+        # Specifically for hyper runs, iteration 0 is the parent run, so it needs the get the artifacts
+        # of the real best iteration which has some other iteration number.
```

Comment:
Changed it a bit but it is still not super-understood.
```suggestion
        # Specifically for hyper runs, iteration 0 is the parent run, so we need to get the artifacts
        # of the real best iteration run, which can be from a different iteration.
```